File: Python/EstruturaDeRepeticao/exercicio_17.py
"""
Faça um programa que calcule o fatorial de um número inteiro fornecido pelo usuário. Ex.: 5!=5.4.3.2.1=120
"""

fatorial = int(input(f"Digite o fatorial: "))
if fatorial <= 0:
    print(f"Não é possível fazer o fatorial de {fatorial}.")
    exit()

print(f"{fatorial}!=", end="")
numero = 1
for i in range(fatorial, 0, -1):
    numero *= i
    if i == 1:
        print(f"{i}", end="")
    else:
        print(f"{i}.", end="")
print(f"={numero}")

File: Java/EstruturaDeRepeticao/Exercicio17.java
// Faça um programa que calcule o fatorial de um número inteiro fornecido pelo usuário. Ex.: 5!=5.4.3.2.1=120
package EstruturaDeRepeticao;
import java.util.Scanner;

public class Exercicio17 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite um número inteiro para calcular o fatorial: ");
        int numero = scanner.nextInt();

        int fatorial = 1;

        System.out.print(numero + "!=");
        for (int i = numero; i >= 1; i--) {
            if (i == 1) {
                System.out.print(i);
            } else {
                System.out.print(i+".");
            }
            fatorial *= i;
        }

        System.out.print("=" + fatorial);

        scanner.close();
    }
}
